includes/class-wc-stripe-order.php

```diff
@@ -0,0 +1,896 @@
+	 * @return void
+	 */
+	public function set_status_before_hold( $status ) {
+		if ( 'default_payment_complete' === $status ) {
+			$payment_complete_status = $this->needs_processing() ? 'processing' : 'completed';
+			$status                  = apply_filters( 'woocommerce_payment_complete_order_status', $payment_complete_status, $this->get_id(), $this );
+		}
+
+		$this->update_meta_data( '_stripe_status_before_hold', $status );
+	}
+
+	/**
+	 * Helper method to retrieve the status of the order before it was put on hold.
+	 *
+	 * @return string The status of the order before it was put on hold.
+	 */
+	public function get_status_before_hold() {
+		$before_hold_status = $this->get_meta( self::META_STRIPE_STATUS_BEFORE_HOLD );
+
+		if ( ! empty( $before_hold_status ) ) {
+			return $before_hold_status;
+		}
+
+		$default_before_hold_status = $this->needs_processing() ? 'processing' : 'completed';
+		return apply_filters( 'woocommerce_payment_complete_order_status', $default_before_hold_status, $this->get_id(), $this );
+	}
+
+	/**
+	 * Set the UPE waiting for redirect flag.
+	 *
+	 * @param $value bool The value to set.
+	 * @return void
+	 */
+	public function set_upe_waiting_for_redirect( $value ) {
+		$this->update_meta_data( self::META_STRIPE_UPE_WAITING_FOR_REDIRECT, $value );
+	}
+
+	/**
+	 * Whether the UPE payment is waiting for redirect.
+	 *
+	 * @return bool
+	 */
+	public function is_upe_waiting_for_redirect() {
+		return (bool) $this->get_meta( self::META_STRIPE_UPE_WAITING_FOR_REDIRECT );
+	}
+
+	/**
+	 * Set the mandate ID.
+	 *
+	 * @param $mandate_id string The mandate ID to set.
+	 * @return void
+	 */
+	public function set_mandate_id( $mandate_id ) {
+		$this->update_meta_data( self::META_STRIPE_MANDATE_ID, $mandate_id );
+	}
+
+	/**
+	 * Get the mandate ID.
+	 *
+	 * @return string
+	 */
+	public function get_mandate_id() {
+		return $this->get_meta( self::META_STRIPE_MANDATE_ID );
+	}
+
+	/**
+	 * Locks an order for payment intent processing for 5 minutes.
+	 *
+	 * @param stdClass $intent The intent that is being processed.
+	 * @return bool            A flag that indicates whether the order is already locked.
+	 */
+	public function lock_payment( $intent = null ) {
+		$this->read_meta_data( true );
+
+		$existing_lock = $this->get_lock_payment();
+
+		if ( $existing_lock ) {
+			$parts         = explode( '|', $existing_lock ); // Format is: "{expiry_timestamp}" or "{expiry_timestamp}|{pi_xxxx}" if an intent is passed.
+			$expiration    = (int) $parts[0];
+			$locked_intent = ! empty( $parts[1] ) ? $parts[1] : '';
+
+			// If the lock is still active, return true.
+			if ( time() <= $expiration && ( empty( $intent ) || empty( $locked_intent ) || ( $intent->id ?? '' ) === $locked_intent ) ) {
+				return true;
+			}
+		}
+
+		$new_lock = ( time() + 5 * MINUTE_IN_SECONDS ) . ( isset( $intent->id ) ? '|' . $intent->id : '' );
```

Comment:
```suggestion
		$new_lock = ( self::PAYMENT_LOCK_EXPIRATION ) . ( isset( $intent->id ) ? '|' . $intent->id : '' );
```

Reply:
Great suggestion, Malith! I have added the constants in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4080/commits/219d20c5e02b61989b550ede1ca7272e410dbff9. Accepted your previous suggestion in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4080/commits/876362e2bd94b82e00435276e9ad03d5165fab10 and adopted this change with a smaller adaptation in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4080/commits/ac262e6d0bbf7cc38a476c54d373178965d9dcf4

Reply:
Perfect. Thanks for catching the mistake in my suggestion and fixing it in https://github.com/woocommerce/woocommerce-gateway-stripe/commit/ac262e6d0bbf7cc38a476c54d373178965d9dcf4